panel/widgets/tables.py

```diff
@@ -1562,6 +1562,11 @@ def _update_max_page(self):
         for ref, (model, _) in self._models.items():
             self._apply_update([], {'max_page': max_page}, model, ref)
 
+    @param.depends("value", watch=True)
+    def _clear_selection_remote_pagination(self):
```

Comment:
Think this may have to be a watcher that checks `if event.new is not event.old`.